lib/cache.js

```diff
@@ -3,144 +3,191 @@ const Tree = require('functional-red-black-tree')
 const Account = require('ethereumjs-account')
 const async = require('async')
 
-var Cache = module.exports = function (trie) {
-  this._cache = Tree()
-  this._checkpoints = []
-  this._trie = trie
-}
-
-Cache.prototype.put = function (key, val, fromTrie) {
-  var modified = !fromTrie
-  this._update(key, val, modified, false)
-}
-
-// returns the queried account or an empty account
-Cache.prototype.get = function (key) {
-  var account = this.lookup(key)
-  if (!account) {
-    account = new Account()
+module.exports = class Cache {
+  constructor (trie) {
+    this._cache = Tree()
+    this._checkpoints = []
+    this._trie = trie
   }
-  return account
-}
 
-// returns the queried account or undefined
-Cache.prototype.lookup = function (key) {
-  key = key.toString('hex')
+  /**
+   * Puts account to cache under its address.
+   * @param {Buffer} key - Address of account
+   * @param {Account} val - Account
+   * @param {bool} [fromTrie]
+   */
+  put (key, val, fromTrie = false) {
+    const modified = !fromTrie
+    this._update(key, val, modified, false)
+  }
 
-  var it = this._cache.find(key)
-  if (it.node) {
-    var account = new Account(it.value.val)
+  /**
+   * Returns the queried account or an empty account.
+   * @param {Buffer} key - Address of account
+   */
+  get (key) {
+    let account = this.lookup(key)
+    if (!account) {
+      account = new Account()
+    }
     return account
   }
```

Comment:
`put` and `get` equivalent, ok, like this explicit addition of the default `false` value for `fromTrie`.